sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignature.cs

```diff
@@ -0,0 +1,435 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Globalization;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+using Azure.Messaging.EventHubs.Core;
+
+namespace Azure.Messaging.EventHubs.Authorization
+{
+    /// <summary>
+    ///   A shared access signature, which can be used for authorization to an Event Hubs namespace
+    ///   or a specific Event Hub.
+    /// </summary>
+    ///
+    public class SharedAccessSignature
```

Comment:
This type adds a lot of public APIs, can we limit the amount or make it non-public?

Reply:
Yeah, I thought that I had flipped this to internal.   It should be, at least for preview.